python/orca/src/bigdl/orca/learn/pytorch/callbacks/tensorboard.py

```diff
@@ -0,0 +1,134 @@
+        super().__init__()
+
+    def on_batch_begin(self, batch):
+        """
+        Called at the beginning of a training batch in `fit` methods.
+        Subclasses should override for any actions to run.
+        @param batch: Integer, index of batch within the current epoch.
+        """
+        pass
+
+    def on_batch_end(self, batch):
+        """
+        Called at the end of a training batch in `fit` methods.
+        Subclasses should override for any actions to run.
+        @param batch: Integer, index of batch within the current epoch.
+        """
+        if self.freq != "epoch" and self._is_rank_zero():
+            if self.freq == "batch" or batch % int(self.freq) == 0:
+                writer = SummaryWriter(log_dir=self.tmp_dir, **self.kwargs)
+                for name, value in logs.items():
+                    if name not in self.unlog_items:
+                        writer.add_scalar(name, value, epoch)
+                writer.close()
+
+    def on_epoch_begin(self, epoch):
+        """
+        Called at the start of an epoch.
+        Subclasses should override for any actions to run. This function should only
+        be called during TRAIN mode.
+        @param epoch: Integer, index of epoch.
+        @param logs: Dict. Currently, saved stats in last epoch has been passed to this argument
+        for this method but may change in the future.
+        """
+        pass
+
+    def on_epoch_end(self, epoch, logs=None):
+        """
+        Called at the end of an epoch.
+        Subclasses should override for any actions to run. This function should only
+        be called during TRAIN mode.
+        @param epoch:  Integer, index of epoch.
+        @param logs: Dict, metric results for this training epoch, and for the validation epoch if
+            validation is performed. Validation result keys are prefixed with val_. For training
+            epoch, the values of the Model's metrics are returned.
+            Example : {'loss': 0.2, 'accuracy': 0.7}
+        """
+        if self.freq == "epoch" and self._is_rank_zero():
+            writer = SummaryWriter(log_dir=self.tmp_dir, **self.kwargs)
+            for name, value in logs.items():
+                if name not in self.unlog_items:
+                    writer.add_scalar(name, value, epoch)
+            writer.close()
+
+    def on_train_begin(self):
+        """
+        Called at the beginning of training.
+        Subclasses should override for any actions to run.
+        @param logs: Dict. Currently, no data is passed to this argument for this method
+          but that may change in the future.
+        """
+        pass
+
+    def on_train_end(self, logs=None):
+        """
+        Called at the end of training.
+        Subclasses should override for any actions to run.
+        """
+        if self._is_rank_zero():
+            put_local_dir_tree_to_remote(self.tmp_dir, self.log_dir)
```

Comment:
add `if not local` here?

Reply:
Such situation will be addressed in `put_local_dir_tree_to_remote`, refer to [link](https://github.com/intel-analytics/BigDL/blob/eae520693fa9b542d1f77787fe1c9bea4cb65ec7/python/orca/src/bigdl/orca/data/file.py#L323).

Reply:
we need to support save to local path here? so maybe better to change as 
```python
if is_local_path(...):
    ...
else:
    put_local_dir_tree_to_remote()
```